test/tests/pbs_smoketest.py

```diff
@@ -335,10 +335,16 @@ def test_finished_jobs(self):
         self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
         a = {'job_history_enable': 'True'}
         self.server.manager(MGR_CMD_SET, SERVER, a)
-        a = {'Resource_List.ncpus': 2}
+        shell_path = os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                                  "bin", "pbs_python")
+        a = {'Resource_List.ncpus': 2, ATTR_S: shell_path}
         j = Job(TEST_USER, a)
         j.set_sleep_time(15)
-        j.create_eatcpu_job(15)
+        mom = self.moms.values()[0].shortname
+        server = self.server.shortname
+        if server == mom:
```

Comment:
just move the localhost check inside create_eatcpu_job() itself, that way PTL can take care of this for the test writer

Reply:
I too agree that these checks can be moved to create_eatcpu_job() but I was unable to access server or mom attributes from Job class, that is the reason i have added the code here

Reply:
you can pass self.server as a argument like you are passing self.mom